index.js

```diff
@@ -20,4 +21,61 @@ window.onclick = function windownone(event) {
   if (event.target === modal) {
     modal.style.display = 'none';
   }
-};
\ No newline at end of file
+};
+
+// Works Modal Popup Window
+// Get Works Section Button Elements
+const work1 = document.getElementById('work1');
+const work2 = document.getElementById('work2');
+const work3 = document.getElementById('work3');
+const work4 = document.getElementById('work4');
+
+// onclick event for each button
+work1.onclick = function work1() {
+  const work1Modal = document.getElementById('work1Modal');
+  work1Modal.style.display = 'block';
+  body.style.overflow = 'hidden';
+};
+work2.onclick = function work2() {
+  const work2Modal = document.getElementById('work2Modal');
+  work2Modal.style.display = 'block';
+  body.style.overflow = 'hidden';
+};
+work3.onclick = function work3() {
+  const work3Modal = document.getElementById('work3Modal');
+  work3Modal.style.display = 'block';
+  body.style.overflow = 'hidden';
+};
+work4.onclick = function work4() {
+  const work4Modal = document.getElementById('work4Modal');
+  work4Modal.style.display = 'block';
+  body.style.overflow = 'hidden';
+};
+
+// Works Modal Close button
+const close1 = document.getElementById('Work1Close');
+const close2 = document.getElementById('Work2Close');
+const close3 = document.getElementById('Work3Close');
+const close4 = document.getElementById('Work4Close');
+
+// onclick event for each close button
+close1.onclick = function close1() {
+  const work1Modal = document.getElementById('work1Modal');
+  work1Modal.style.display = 'none';
+  body.style.overflow = 'visible';
+};
+close2.onclick = function close2() {
+  const work2Modal = document.getElementById('work2Modal');
+  work2Modal.style.display = 'none';
+  body.style.overflow = 'visible';
+};
+close3.onclick = function close3() {
+  const work3Modal = document.getElementById('work3Modal');
+  work3Modal.style.display = 'none';
+  body.style.overflow = 'visible';
+};
+close4.onclick = function close4() {
+  const work4Modal = document.getElementById('work4Modal');
+  work4Modal.style.display = 'none';
+  body.style.overflow = 'visible';
+};
```

Comment:
- [ ] You have done a tremendous job in both styling and writing the JavaScript logic for making your portfolio popup show. However, kindly note that according to the [project requirements](https://github.com/microverseinc/curriculum-html-css/blob/main/portfolio/8_popup.md), you are required to use an array to store the information of the popup. Kindly make this change so that you stick to the project requirements.

**I figured out that you may have some challenge implementing this requirement, so I decided to show you practically how to do it with a very simple and basic example using arrays to store your data:**

**Index.html**

```
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Document</title>
</head>
<body>
  
  <h1>Content section</h1>
  <section class="lists"></section>
  <script src="./script.js"></script>
</body>
</html>
```

**script.js**

```
const sect = document.querySelector('.lists');

const my_lists = [
  {
    id: 1,
    heading: 'My heading',
    desc: 'My desc'
  },
  {
    id: 2,
    heading: 'My heading two',
    desc: 'My desc'
    
  },
  {
    id: 3,
    heading: 'My heading three',
    desc: 'My desc'
  }
];

let allLists = () => {
  sect.innerHTML = '';
  for(let i = 0; i < my_lists.length; i++) {
    const listWrapper = document.createElement('div');
    const listHeading = document.createElement('h2');
    listHeading.textContent = my_lists[i].heading;
    const listDesc = document.createElement('p');
    listDesc.textContent = my_lists[i].desc;
    const listBtn = document.createElement('button');
    listBtn.textContent = 'Details';

    listWrapper.append(listHeading, listDesc, listBtn);

    sect.appendChild(listWrapper)

    listBtn.addEventListener('click', () => {
      listPopup(my_lists[i].id)
    })
  }

}


window.addEventListener('load', () => {
  allLists()
})
```

The idea behind this is to have your sections dynamically generated by JavaScript. When the page loads, I am looping through my whole array and dynamically printing out the contents and displaying them to the screen. You could optionally use the JavaScript [template literal approach](https://www.w3schools.com/js/js_string_templates.asp) but this will quicky bring you a problem when you want to attach the event listener for the button to bring up your popup. 

**This is what I get when I open the project on the browser:**

![image](https://user-images.githubusercontent.com/86318284/180009915-d3bcf574-7a6c-4c18-9a68-cbeb79e99f5a.png)

Reply:
thank you so much for the startup code, it helped greatly
i have misread the project requirements at first which has result me in doing my project in a wrong way

Reply:
Alright @rebaz36 . You are most welcome.